### PR TITLE
Update free-programming-books-fr.md (add missing creators, formats, typo, et al.)

### PR DESCRIPTION
**Pull request by Mamadou442**
## What does this PR do?
Improve repo

### Description

- change a few links of books to HTTPS.
- It specifies the creators, formats of a few books.
- change "Histoires et cultures du Libre" as archived books.
- fix typo ("Activités débranchées")

## Checklist:
- [x] Read our [contributing guidelines](https://github.com/EbookFoundation/free-programming-books/blob/main/docs/CONTRIBUTING.md).
- [x] [Search](https://ebookfoundation.github.io/free-programming-books-search/) for duplicates.
- [x] Include author(s) and platform where appropriate.
- [x] Put lists in alphabetical order, correct spacing.
- [x] Add needed indications (PDF, access notes, under construction).
- [x] Used an informative name for this pull request.

## Follow-up

- Check the status of GitHub Actions and resolve any reported warnings!

### CODE REVIEW

File: books/free-programming-books-fr.md [hunk 1]
```diff
@@ -317,12 +317,12 @@
...
 ### Scratch
 
-* [Informatique Créative](https://pixees.fr/programmation-creative-en-scratch/)
+* [Informatique Créative](https://pixees.fr/programmation-creative-en-scratch/) - Christan Balch, Michelle Chung, Karen Brennan, `trl.:` INRIA, PROVENCE TRADUCTION (PDF, PPTX)
 
 
 ### SPIP
 
-* [Programmer avec SPIP](http://programmer.spip.net) - Matthieu Marcimat, collectif SPIP
+* [Programmer avec SPIP](https://programmer.spip.net) - Matthieu Marcimat, collectif SPIP (HTML)
```
Comment (eshellman) [inline books/free-programming-books-fr.md:325]:
```suggestion
* [Programmer avec SPIP](https://programmer.spip.net) - Matthieu Marcimat, collectif SPIP
```

File: books/free-programming-books-fr.md [hunk 2]
```diff
@@ -317,12 +317,12 @@
 
 ### Scratch
 
-* [Informatique Créative](https://pixees.fr/programmation-creative-en-scratch/)
+* [Informatique Créative](https://pixees.fr/programmation-creative-en-scratch/) - Christan Balch, Michelle Chung, Karen Brennan, `trl.:` INRIA, PROVENCE TRADUCTION (PDF, PPTX)
```
Comment (eshellman) [inline books/free-programming-books-fr.md:320]: no ALL CAPS

File: books/free-programming-books-fr.md [hunk 3]
```diff
@@ -184,12 +184,12 @@
 ### Haskell
 
 * [A Gentle Introduction to Haskell](http://gorgonite.developpez.com/livres/traductions/haskell/gentle-haskell/) - Paul Hudak, John Peterson, Joseph Fasel, `trl.:` Nicolas Vallée, Gnux, ggnore, fearyourself, Joyeux-oli, Kikof, khayyam90
-* [Apprendre Haskell vous fera le plus grand bien !](http://lyah.haskell.fr)
+* [Apprendre Haskell vous fera le plus grand bien !](https://lyah.haskell.fr) - Miran Lipovača, `trl.:` Valentin Robert (HTML, PDF)
```
Comment (eshellman) [inline books/free-programming-books-fr.md:187]:
```suggestion
* [Apprendre Haskell vous fera le plus grand bien !](https://lyah.haskell.fr) - Miran Lipovača, `trl.:` Valentin Robert
```
don't need format note id HTML or multiple

File: books/free-programming-books-fr.md [hunk 4]
```diff
@@ -113,7 +113,7 @@
 
 ### Bash / Shell
 
-* [Guide avancé d'écriture des scripts Bash](http://abs.traduc.org/abs-fr/)
+* [Guide avancé d'écriture des scripts Bash](https://abs.traduc.org/abs-fr/) - Mendel Cooper, `trl.:` Adrien Rebollo et al. (HTML)
```
Comment (eshellman) [inline books/free-programming-books-fr.md:116]:
```suggestion
* [Guide avancé d'écriture des scripts Bash](https://abs.traduc.org/abs-fr/) - Mendel Cooper, `trl.:` Adrien Rebollo et al.
```

File: books/free-programming-books-fr.md [hunk 5]
```diff
@@ -76,7 +76,7 @@
 
 #### Logiciels libres
 
-* [Histoires et cultures du Libre](http://framabook.org/histoiresetculturesdulibre/)
+* [Histoires et cultures du Libre](https://archives.framabook.org/histoiresetculturesdulibre/) - Camille Paloque-Berges, Christophe Masutti, `edt.:` Framasoft (coll. Framabook) [(HTML, PDF, EPUB, et al.)] *(:card_file_box: archived)*
```
Comment (eshellman) [inline books/free-programming-books-fr.md:79]:
```suggestion
* [Histoires et cultures du Libre](https://archives.framabook.org/histoiresetculturesdulibre/) - Camille Paloque-Berges, Christophe Masutti, `edt.:` Framasoft (coll. Framabook)
```
we use the 'archived' notation for internet archive style archiving

**Review by eshellman — CHANGES_REQUESTED** [review]
just some tweaks, thanks!

**Comment by Mamadou442** [comment]
Tried to add all of the tweaks in a single commit. Thank for your review!

**Pull request merged**